tests/tracing/instrumentation/test_action_executor.py

```diff
@@ -88,3 +114,72 @@ def test_instrument_action_executor_run_registers_tracer(
 
     assert tracer is not None
     assert tracer == mock_tracer
+
+
+@pytest.mark.parametrize(
+    "events, get_dispatcher, expected",
+    [
+        (
+            [],
+            get_dispatcher0,
+            {"events": "[]", "slots": "[]", "utters": "[]", "message_count": 0},
+        ),
+        (
+            [ActionExecuted("my_form")],
+            get_dispatcher2,
+            {"events": '["action"]', "slots": "[]", "utters": "[]", "message_count": 1},
+        ),
+        (
+            [ActionExecuted("my_form"), SlotSet("my_slot", "some_value")],
+            get_dispatcher1,
+            {
+                "events": '["action", "slot"]',
+                "slots": '["my_slot"]',
+                "utters": '["utter_greet"]',
+                "message_count": 1,
+            },
+        ),
+        (
+            [SlotSet("my_slot", "some_value")],
+            get_dispatcher3,
+            {
+                "events": '["slot"]',
+                "slots": '["my_slot"]',
+                "utters": '["utter_greet"]',
+                "message_count": 2,
+            },
+        ),
+    ],
+)
+def test_tracing_action_executor_create_api_response(
+    tracer_provider: TracerProvider,
+    span_exporter: InMemorySpanExporter,
+    previous_num_captured_spans: int,
+    events: Optional[List],
+    get_dispatcher: callable,
```

Comment:
```suggestion
    get_dispatcher: Callable,
```
should be imported from `typing` too.